### ISSUE TITLE
Consider automating the creation of the Python version used

### ISSUE THREAD

**Issue by nolangormley** [cmu-delphi/covidcast-indicators#1854]
Currently, running these indicators requires that you have Python3.8 installed locally. As 3.8 is a fairly old version, it is no longer in the standard Ubuntu libraries and needs to be installed from binaries. We should consider new options that allow us to make this easier in the future.

One proposed option would be to have a script in the root directory that will download and install the python3.8 binaries like [this](https://tecadmin.net/install-python-3-8-ubuntu/)

At the very least we should have some build instructions in the main readme about this requirement.

**Comment by dshemetov** [comment]
Installing Python in 2023 feels so janky. FWIW, I heard [`deadsnakes` is a well-maintained PPA for this](https://launchpad.net/~deadsnakes/+archive/ubuntu/ppa). I _think_ it's legit.

Also, an alternative may be to bump our Python version to 3.9 or 3.10, here. I don't think there's much backwards incompatibility in these versions.

**Comment by nolangormley** [comment]
I did try the deadsnakes ppa, not sure if I just flubbed something but I still couldn't get 3.8 through term. 

I usually use Pyenv to manage Python versions, but didn't have it configured on my laptop. Do you use a solution like that when you run these indicators locally?

**Comment by dshemetov** [comment]
Dang, that's too bad. I haven't actually tried deadsnakes, only read about it recently. And yea, I'm still using pyenv, it just feels so jank having to install a third-party tool, mess with your .zshrc or whatever, and then install Python.

**Comment by nolangormley** [comment]
Agree. Not sure if there's an elegant solution tbh, just thought I would put this issue up to console my 20 minutes of frustration lol

**Comment by krivard** [comment]
Settling on a new version of python is a big lift we should be prepared to dedicate serious time toward it if we find surprises.

In the meantime, `pyenv` looks to be the way people manage multiple weird versions of python these days. I think we use it in our prod systems as well. May be worth converting the docs and indicator build scripts to use it?

A sketch of the changes needed might look like this:

# getting started guide

1. install pyenv for your OS: https://github.com/pyenv/pyenv#installation

# Makefile

venv:
1. use `pyenv` to check whether 3.8 is installed
2. if not, `pyenv install 3.8`
3. use `pyenv` to switch to 3.8 before creating the new environment

& then I think all targets that source env/bin/activate should just... work
